api/rest/bundle_handler.go

```diff
@@ -56,14 +56,31 @@ type realClock struct{}
 
 func (realClock) Now() time.Time { return time.Now() }
 
-func NewBundleHandler(workDir string, collectors []collector.Collector, timeout time.Duration) BundleHandler {
-	return BundleHandler{
+func NewBundleHandler(workDir string, collectors []collector.Collector, timeout time.Duration) (*BundleHandler, error) {
+	f, err := os.Stat(workDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			err = os.Mkdir(workDir, dirPerm)
+			if err != nil {
+				logrus.WithError(err).Errorf("workDir does not exist and could not be created %s", workDir)
+				return nil, err
+			}
+		} else {
+			logrus.Errorf("Could not open workDir %s", workDir)
+			return nil, err
+		}
+	} else if f.Mode().IsRegular() {
+		logrus.Errorf("workDir exists but is not a directory: %s", workDir)
+		return nil, err
+	}
```

Comment:
How about replacing the code above with


```go
err := os.MkdirAll(workDir, dirPerm)
if err != nil {
	return nil, fmt.Errorf("could not create %s: %s", workDir, err)
}
```

Reply:
> MkdirAll creates a directory named path, along with any necessary parents, and returns nil, or else returns an error. The permission bits perm (before umask) are used for all directories that MkdirAll creates. If path is already a directory, MkdirAll does nothing and returns nil.

Nice, ya that seems to be a good idea @janisz